pkg/node/types/node.go

```diff
@@ -223,6 +223,26 @@ func (n *Node) getNodeIP(ipv6 bool) (net.IP, addressing.AddressType) {
 	return backupIP, ipType
 }
 
+// GetK8sNodeIPs returns k8s Node IP (either InternalIP or ExternalIP or nil;
+// the former is prefered).
+func (n *Node) GetK8sNodeIP() net.IP {
+	var externalIP net.IP
+
+	for _, addr := range n.IPAddresses {
+		if addr.Type == addressing.NodeInternalIP {
+			return addr.IP
+		} else if addr.Type == addressing.NodeExternalIP {
+			externalIP = addr.IP
+		}
+	}
+
+	if externalIP != nil {
+		return externalIP
+	}
+
+	return nil
```

Comment:
can be `return externalIP`

Reply:
I wanted to make it more visible that `nil` can be returned as well.

Reply:
Doesn't the function comment already describe this? And the declaration of the stack implies potential lack of initialization? (I don't overly care)